cli/replicaset/configsource_test.go

```diff
@@ -723,3 +748,98 @@ func TestCConfigSource_Expel_single_key(t *testing.T) {
 `)
 	assertPublished(t, publisher, "a", revision, expected)
 }
+
+func TestCConfigSource_AddRole(t *testing.T) {
+	cfg := []byte(`groups:
+  group-1:
+    replicasets:
+      replicaset-001:
+        instances:
+          instance-001:
+            iproto:
+              listen: {}`)
+	type tCase struct {
+		name        string
+		rolesAddCtx replicaset.RolesAddCtx
+		cfg         []byte
+		expectedCfg []byte
+		isErr       bool
+		errMsg      string
+	}
+	cases := []tCase{
+		{
+			name: "ok",
```

Comment:
The same, it's better to test the function on different scopes.

Reply:
Added cases.